src/tike/ptycho/ptycho.py

```diff
@@ -167,12 +167,7 @@ def reconstruct(
             logger.info("{} for {:,d} - {:,d} by {:,d} frames for {:,d} "
                         "iterations.".format(algorithm, *data.shape[1:],
                                              num_iter))
-            scan, data = asarray_multi_split(
-                operator,
-                pool.num_workers,
-                scan,
-                data,
-            )
+            scan, data = split_by_scan_stripes(pool, scan, data, operator.fly)
```

Comment:
I suggest setting the splitting scheme as an input parameter (e.g. 'grid' and 'stripe') and use an if condition statement here to let the users freely select the splitting approach.

Reply:
There is now an input parameter for switching between `grid` and stripe. The stripe method is default because the grid method only support specific GPU device pools (when the primary GPU is zero and the total number of GPUs is a power of 2).